sympy/matrices/expressions/diagonal.py

```diff
@@ -1,14 +1,19 @@
 from __future__ import print_function, division
 
 from sympy.matrices.expressions import MatrixExpr
-from sympy.core import S
-
+from sympy.core import S, Eq
+from sympy.functions.special.tensor_functions import KroneckerDelta
 class DiagonalMatrix(MatrixExpr):
     arg = property(lambda self: self.args[0])
     shape = property(lambda self: (self.arg.shape[0], self.arg.shape[0]))
 
     def _entry(self, i, j):
-        return S.Zero if i != j else self.arg[i, 0]
+        eq = Eq(i, j)
+        if eq is S.false:
+            return S.Zero
+        elif eq is S.true:
+            return self.arg[i, 0]
```

Comment:
@smichr 
If I take your advice then It will fail this test:-
```assert D[1, 1] == x[1, 0]```
Travis CI build test is failing because it is showing time limit exceeded but everything was working fine before. I am not failing tests, It's showing time limit ecxeeded even though all the tests were getting evaluated but it just stopped and gave time limit exceeded And also it is just one test.
Right now I am reverting back to when everything was working fine and only issues of spaces were there. Either the test is wrong or i.e. we should remove it and add new one or I can do something else, the thing is ,this issue started when I made 2 commits to this branch one after another very quickly the time diifference between them was like 10 seconds this caused my first test to fail because it got interrupted. I think something went wrong there. I am not sure.